index.js

```diff
@@ -30,13 +30,15 @@ module.exports = function block (size, opts) {
     bufferedBytes += data.length
     buffered.push(data)
 
+    var b = Buffer.concat(buffered)
```

Comment:
we canavoid this allocation as well, by just queuing new buffers directly from the array I think

Reply:
actually we can't, never mind. the overhead of slicing into the buffers wouldn't be worth it probably

Reply:
Hm. It's probably only worth looking at with a specific test case where it suffers from that.

In general I assume the new worst case scenario will be when you require 2 buffers from your source on average to create one buffer out and are receiving a high volume of buffers. For example 130B buffers in and 250B buffers out. For every pair it would concat once, then slice twice.

The only way I see that improving is by tracking the buffer offsets as well and keeping a little more data in the cache than strictly necessary. Never using slice or concat at all, but allocating the full output buffer at once and doing (partial) copies into that.